### ISSUE TITLE
Warning in the error log

### ISSUE THREAD

**Issue by Zodiac1978** [pluginkollektiv/antispam-bee#508]
**Describe the bug**

There are php warning the error log with PHP 8.1 (first reported in the [forums](https://wordpress.org/support/topic/php-8-1-notices-3/))

```
    [20-Jun-2023 20:22:23 UTC] PHP Warning: Trying to access array offset on value of type null in /home/wp-content/plugins/antispam-bee/antispam_bee.php on line 2626
    [20-Jun-2023 20:22:23 UTC] PHP Warning: Undefined array key “reasons” in /home/wp-content/plugins/antispam-bee/antispam_bee.php on line 2626
```

Here is a link to the line:
https://github.com/pluginkollektiv/antispam-bee/blob/master/antispam_bee.php#L2626

I'm afraid this is another consequence of moving the defaults to its own function in https://github.com/pluginkollektiv/antispam-bee/pull/478/commits/2cd7a991c973dc26474841c5fccd66a2b9ddd19b

@florianbrinkmann Can you have a look at this?

**Comment by florianbrinkmann** [comment]
@Zodiac1978 I created a PR that should fix the issue (#517)

**Comment by Zodiac1978** [comment]
Fixed with #517
